Added index failure checks

theherodied opened the pull request:
Added index failure checks using -w for warning and -c for critical.
Added contributers = "kahluagenie, theherodied"

Example: If index failures exceeds critical limit it will return:
+     $ ./check_graylog2 -l http://localhost:12900 -u USERNAME -p PASSWORD -w 10 -c 20
+    CRITICAL - Index Failure above Critical Limit!
+    Service is running
+    533732628 total events processed
+    21 index failures
+    297 throughput
+    1 sources
+    Check took 94ms
+    |time=0.0094;;;; total=533732628;;;; sources=1;;;; throughput=297;;;; index_failures=21;;;;

theherodied committed:
Added -w and -c index checks

Added -w and -c for index warning and index critical limits.

theherodied committed:
Add -w and -c index checks

Added -w and -c for index warning and index critical limits.

theherodied committed:
Added -w and -c index checks

Added -w and -c for index warning and index critical limits.

theherodied committed:
Added -w and -c index limits

Added -w and -c for index warning and index critical limits.

catinello reviewed (changes requested):
Please add the quit function to the error handling for the strconv.ParseFloat calls.

quit(UNKNOWN, "Can not parse index warning errors.", err)
quit(UNKNOWN, "Can not parse index critical errors.", err)

The rest lgtm! Thanks for your PR.

theherodied committed:
Added  quit function to strconv calls

Please add the quit function to the error handling for the strconv.ParseFloat calls.

quit(UNKNOWN, "Can not parse index warning errors.", err)
quit(UNKNOWN, "Can not parse index critical errors.", err)

theherodied committed:
added strconv error handling, corrected comment

theherodied commented:
Good catch. I think it's good now as of commit 8104708.
Let me know if anything needs attention.

theherodied commented:
@catinello, I'm going to work on querying the api for X number of index errors within a specified time. I.E. 24H, 48H, etc... that can be specified from the command line with something like -t 24.

Thoughts?

catinello commented:
@theherodied I'm definitely interested to see your solution, not sure if it's the right place though.

You may want to solve this in the monitoring tool. Because this opens up questions to your check interval schedule, alerting schemes and the way that your check provides results. 

You can set a fixed number of errors and time period, but for how long will you hold that status? What if you reach the threshold just before the time runs out. Do you want to be notified? Or what if you have reached the amount of errors after your first check because of a short outage, do you want to get a notice x-times for the next 24/48 Hours? 

You're going to run into the issue of storing the temp. value to some rw location/database. Not a trivial task to do while staying compatible as a simple nagios check plugin.
